frontend/server/app.ts

```diff
@@ -139,8 +139,29 @@ function createUIServer(options: UIConfigs) {
   registerHandler(app.delete, '/apps/tensorboard', tensorboardDeleteHandler);
   registerHandler(app.post, '/apps/tensorboard', tensorboardCreateHandler);
 
-  /** Pod logs */
-  registerHandler(app.get, '/k8s/pod/logs', getPodLogsHandler(options.argo, options.artifacts));
+  /** Pod logs - conditionally stream through API server, otherwise directly from k8s and archive */
+  if (options.artifacts.streamLogsFromServerApi) {
+    app.all(
+      '/k8s/pod/logs',
+      proxy({
+        changeOrigin: true,
+        onProxyReq: proxyReq => {
+          console.log('Proxied log request: ', proxyReq.path);
+        },
+        headers: HACK_FIX_HPM_PARTIAL_RESPONSE_HEADERS,
+        pathRewrite: (pathStr: string, req: any) => {
+          /** Argo nodeId is just POD name */
+          const nodeId = req.query.podname;
+          const runId = req.query.runid;
+          return `/${apiVersionPrefix}/runs/${runId}/nodes/${nodeId}/log`;
+        },
+        target: apiServerAddress,
+      }),
+    );
+  } else {
+    registerHandler(app.get, '/k8s/pod/logs', getPodLogsHandler(options.argo, options.artifacts));
```

Comment:
If this API is implemented in kfp api server, I think we can deprecate this part as  it is essentially identical to the golang version except it doesn't handle the log formats (it just stream outright - but not difficult to add a transform to the stream thou).

Reply:
@eterna2 that's correct. We can deprecate, but we need to remember that streaming through server API means one hop more (UI -> API proxy -> Server API -> k8s API). Even though, I don't see significant performance impact.

Reply:
I don't see a big concern with one more hop too, a lot of requests have this extra hop, but they are also working well.

Reply:
Agreed. Let's do it in the subsequent PR once this is merged.